lib/src/RedisRemoteSaiInterface.cpp

```diff
@@ -2097,3 +2106,20 @@ void RedisRemoteSaiInterface::refreshTableDump()
                 kvp.second.size());
     }
 }
+
+// Scale a channel timeout interval up based on the object count.
+unsigned int RedisRemoteSaiInterface::scale_timeout_interval(_In_ uint32_t object_count)
+{
+    SWSS_LOG_ENTER();
+
+    // We return a value which is the greater of:
+    // - The default 60s timeout interval
+    // - 50ms per object
+    unsigned int scaled_wait = (unsigned int)object_count * PER_OBJ_TIMEOUT;
+    if (scaled_wait > DEFAULT_GETRESPONSE_TIMEOUT_MS)
```

Comment:
You could return `std::max(object_count * PER_OBJ_TIMEOUT, DEFAULT_GETRESPONSE_TIMEOUT_MS);`